### PR TITLE
use the actual unique id to compare connection

### PR DESCRIPTION
**Pull request by alanrenmsft**
This PR fixes https://github.com/microsoft/azuredatastudio-postgresql/issues/113

there are 2 bugs behind this issue:
1. one needs to be made in PG extension, @nasc17 as we chatted offline, please add 'isIdentity= true' to the port option in your package.json, this will make ADS be able to treat them as different connections.
2. Object explorer is not using the actual unique string of connections to compare instead it is using the hardcoded options when looking for existing profiles, I am fixing this problem with this PR.

btw, MSSQL do not have the same issue because we are hardcoding it in ADS core to append the port number after the server name.

### CODE REVIEW

File: src/sql/platform/connection/common/connectionProfile.ts [hunk 1]
```diff
@@ -84,30 +83,14 @@ export class ConnectionProfile extends ProviderConnectionInfo implements interfa
...
 	public static matchesProfile(a: interfaces.IConnectionProfile, b: interfaces.IConnectionProfile): boolean {
-		return a && b
-			&& a.providerName === b.providerName
-			&& ConnectionProfile.nullCheckEqualsIgnoreCase(a.serverName, b.serverName)
-			&& ConnectionProfile.nullCheckEqualsIgnoreCase(a.databaseName, b.databaseName)
-			&& ConnectionProfile.nullCheckEqualsIgnoreCase(a.userName, b.userName)
-			&& ConnectionProfile.nullCheckEqualsIgnoreCase(a.options['databaseDisplayName'], b.options['databaseDisplayName'])
-			&& a.authenticationType === b.authenticationType
-			&& a.groupId === b.groupId;
+		return a && b && a.getOptionsKey() === b.getOptionsKey();
```
Comment (Charles-Gagnon) [inline src/sql/platform/connection/common/connectionProfile.ts:86]: If these can be undefined please update the param to be `interfaces.IConnectionProfile | undefined`

**Review by Charles-Gagnon — APPROVED** [review]

**Comment by alanrenmsft** [comment]
@nasc17 why did you closing this pr? by mistake?

**Comment by nasc17** [comment]
> @nasc17 why did you closing this pr? by mistake?

Yes mistake. When I closed the PRs I had opened tied to this it must have closed here as well.

File: src/sql/platform/connection/test/node/connectionStatusManager.test.ts [hunk 2]
```diff
@@ -251,6 +251,7 @@ suite('SQL ConnectionStatusManager tests', () => {
 		newConnection.options['databaseDisplayName'] = newConnection.databaseName;
 		connections.addConnection(newConnection, 'test_uri_1');
 		connections.addConnection(newConnection, 'test_uri_2');
+		newConnection = new ConnectionProfile(capabilitiesService, newConnection);
```
Comment (alanrenmsft) [inline src/sql/platform/connection/test/node/connectionStatusManager.test.ts:254]: previously the newConnection is a mock object, not an actual ConnectionProfile object, the getOptionsKey of it is mocked to always return 'connection1' which caused the test failure.

**Review by nasc17 — APPROVED** [review]

**Pull request merged**